source/code/util/xml/XMLReader.cpp

```diff
@@ -315,12 +356,12 @@ void XMLReader::_ToCharRef(Utf8String& /*unused*/, char& ch)
     size_t i = 1;
     m_CharStartPos++;
     m_CharPos++;
-    if (*m_CharStartPos == c_x)
+    if (*m_CharStartPos == u8_x)
     {
         // This is hex
         isHex = true;
         // Move one more for x
-        m_CharStartPos++;
+        *m_CharStartPos++;
```

Comment:
May I know why do we need to dereference the iterator?
 I believe  ```*m_CharStartPos++``` parses as  ```*(m_CharStartPos++)```.
  
Earlier we were just incrementing the iterator. Now, we not only increment the iterator, we are also dereferencing it. But then, we are not capturing the return value.